Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,512 @@
+        addSubview(subview)
+        subview.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            subview.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            subview.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+            subview.topAnchor.constraint(equalTo: self.topAnchor),
+            subview.bottomAnchor.constraint(equalTo: self.bottomAnchor)
+            ])
+    }
+    
+    /// Start the locationManager with undetermined access.
+    fileprivate func startWithAccessNotDetermined() {
+        if (Bundle.main.object(forInfoDictionaryKey: "NSLocationWhenInUseUsageDescription") != nil) {
+            locationManager.requestWhenInUseAuthorization()
+        }
+        if (Bundle.main.object(forInfoDictionaryKey: "NSLocationAlwaysUsageDescription") != nil) {
+            locationManager.requestAlwaysAuthorization()
+        }
+        else{
+            didStartOrFailWithError(ArcGISARView.clError.missingPListKey)
+        }
+    }
+    
+    /// Start updating the location and heading via the locationManager.
+    fileprivate func startUpdatingLocationAndHeading() {
+        locationManager.startUpdatingLocation()
+        if CLLocationManager.headingAvailable() {
+            locationManager.startUpdatingHeading()
+        }
+    }
+    
+    /// Stop updating location and heading.
+    fileprivate func stopUpdatingLocationAndHeading() {
+        locationManager.stopUpdatingLocation()
+        if CLLocationManager.headingAvailable() {
+            locationManager.stopUpdatingHeading()
+        }
+    }
+
+    /// Start the locationManager with denied access.
+    fileprivate func startWithAccessDenied() {
+        didStartOrFailWithError(ArcGISARView.clError.accessDenied)
+    }
+    
+    /// Start the locationManager with authorized access.
+    fileprivate func startWithAccessAuthorized() {
+        startUpdatingLocationAndHeading()
+    }
+    
+    /// Potential notification to the user of an error starting device tracking.
+    ///
+    /// - Parameter error: The error that occurred when starting tracking.
+    fileprivate func didStartOrFailWithError(_ error: Error?) {
+        if !notifiedStartOrFailure, let error = error {
+            // TODO: present error to user...
+            print("didStartOrFailWithError: \(String(reflecting:error))")
+        }
+        
+        notifiedStartOrFailure = true
+    }
+    
+    /// Handle a change in authorization status to "denied".
+    fileprivate func handleAuthStatusChangedAccessDenied() {
+        // auth status changed to denied
+        stopUpdatingLocationAndHeading()
+        
+        // We were waiting for user prompt to come back, so notify.
+        didStartOrFailWithError(ArcGISARView.clError.accessDenied)
+    }
+    
+    /// Handle a change in authorization status to "authorized".
+    fileprivate func handleAuthStatusChangedAccessAuthorized() {
```

Comment:
I can see why you want to re-use the location datasource, because it does some of this code for you. However - that is really written to power the location display. It sets it's own location manager up, and it might be setup differently than you need it setup for AR. 

It might be best to just move this out into an extension or another object just for AR?

Reply:
I really would like to get rid of all that code as it was copied from the `AGSCLLocationDataSource`.  There's a big push to use the `LocationDatasource` and the design has already been changed..  The `AGSCLLocationDataSource` has a property for the `CLLocationManager` (`@property (nonatomic, strong, readonly) CLLocationManager *locationManager;`) so I think we'll be OK.

Reply:
I'm starting to feel like the LocationDatasource object to use. if we change the location manager that the map is using, that will have affects on how the map is getting it's locations, which may or may not be desired.

Just my thought right now is CLLocationManager is the object to go to for getting locations, the LocationDatasource...

Reply:
Although it seems that ship has sailed, at least for the moment. So let's ignore my comment for now